scripts/npm-dist-tag.sh

```diff
@@ -1,12 +1,43 @@
 #! /bin/bash
-# npm-dist-tag.sh - add/remove/list dist-tags for NPM registry packages 
-# Try: `npm-dist-tag.sh` for usage.
+usage() { cat <<END_USAGE
+Usage:
+$0 [--dry-run] [lerna] add <tag> [-<pre-release>]
+  Add <tag> to package dist-tags for current version or specified <pre-release>.
+$0 [--dry-run] [lerna] <remove|rm> <tag>
+  Remove <tag> from package dist-tags.
+$0 [--dry-run] [lerna] <list|ls> [<tag>]
+  List package dist-tags, or just the one named <tag>.
+
+With "--dry-run", npm commands are printed to standard error rather than executed.
+
+If the first operand is "lerna", the operation is extended to all packages.
+END_USAGE
+  exit 1
+}
+
+# fail <error message>
+fail () {
+	printf 1>&2 '%s\n\n' "$1"
+	usage 1>&2
+}
```

Comment:
Shell trick: you can do an io-redirect at the end of the function, though some find this too obscure. I say there's no such thing as "too obscure" when shell programming!
```suggestion
	printf '%s\n\n' "$1"
	usage
} 1>&2
```

Reply:
Nice; thanks!